### DISCUSSION TITLE
export WMS layer

### DISCUSSION THREAD

**Discussion by wegerv** [Leaflet/Leaflet#8056] [Q&A]
Hi,

I'm trying to extend leaflet4vaadin (https://vaadin.com/directory/component/leaflet4vaadin) with the leaflet WMS layer (TileLayer.WMS). 
This only succeeds if I export TileLayerWMS like this (for version 1.6.0, but 1.8.0. doesn't export the class either):

exports.TileLayerWMS = TileLayerWMS;
exports.tileLayerWMS = tileLayerWMS; 

If I don't leaflet throws "The provided object is not a Layer" when the WMS layer is added to the map from the java code.

Is there a special reason TileLayer.WMS is not exported like other layers?

Thanks, Vincent

**Comment by IvanSanchez** [comment]
This falls under the "it's the framework's problem" category of problems. If anything, this should be reported as a bug at https://github.com/Gubancs/leaflet4vaadin/issues instead.

Besides, there's also https://github.com/mstahv/v-leaflet and therefore https://github.com/mstahv/v-leaflet/blob/master/src/main/java/org/vaadin/addon/leaflet/LWmsLayer.java , so beware of duplicating efforts.

**Reply by wegerv** [reply]
I don't think so. 

See e.g. the index.js in version 1.8.0.

The question is about the choice for;

export {GridLayer, gridLayer} from './GridLayer';
import {TileLayer, tileLayer} from './TileLayer';
import {TileLayerWMS, tileLayerWMS} from './TileLayer.WMS';
**TileLayer.WMS = TileLayerWMS;
tileLayer.wms = tileLayerWMS;**
export {TileLayer, tileLayer};

instead of using:

export {GridLayer, gridLayer} from './GridLayer';
import {TileLayer, tileLayer} from './TileLayer';
import {TileLayerWMS, tileLayerWMS} from './TileLayer.WMS';
**export {TileLayerWMS, tileLayerWMS};**
export {TileLayer, tileLayer};

**Comment by IvanSanchez** [comment]
It's been like that for five years (meaning: including 1.6.0), see https://github.com/Leaflet/Leaflet/blame/main/src/layer/tile/index.js .
